qcodes/instrument_drivers/tektronix/Keithley_3706A_Matrix_channels.py

```diff
@@ -0,0 +1,600 @@
+                           vals=vals.Strings())
+
+        self.add_parameter('channel_connect_rule',
+                           get_cmd=self._get_channel_connect_rule,
+                           set_cmd=self._set_channel_connect_rule,
+                           docstring='Controls the connection rule for closing '
+                                     'and opening channels when using '
+                                     '`exclusive_close` and '
+                                     '`exclusive_slot_close` parameters. '
+                                     'If it is set to break before make, '
+                                     'it is ensured that all channels open '
+                                     'before any channels close. If it is set '
+                                     'to make before break, it is ensured that '
+                                     'all channels close before any channels '
+                                     'open. If it is off, channels open and '
+                                     'close simultaneously.',
+                           vals=vals.Enum('BREAK_BEFORE_MAKE',
+                                          'MAKE_BEFORE_BREAK',
+                                          'OFF'))
+
+        self.add_parameter('gpib_enable',
+                           get_cmd=self._get_gpib_status,
+                           set_cmd=self._set_gpib_status,
+                           docstring='Enables or disables GPIB connection.',
+                           val_mapping=create_on_off_val_mapping(on_val='true',
+                                                                 off_val='false'
+                                                                 ))
+
+        self.add_parameter('gpib_address',
+                           get_cmd=self._get_gpib_address,
+                           get_parser=int,
+                           set_cmd=self._set_gpib_address,
+                           docstring='Sets and gets the GPIB address.',
+                           vals=vals.Ints(1, 30))
+
+        self.add_parameter('lan_enable',
+                           get_cmd=self._get_lan_status,
+                           set_cmd=self._set_lan_status,
+                           docstring='Enables or disables LAN connection.',
+                           val_mapping=create_on_off_val_mapping(on_val='true',
+                                                                 off_val='false'
+                                                                 ))
+
+        self.connect_message()
+
+    def _reset_channel(self, val: str) -> None:
+        if not self._validator(val):
+            raise InvalidValue(f'{val} is not a valid specifier. '
+                               'The specifier should be channels, channel '
+                               'ranges, slots, backplane relays or "allslots".')
+        self.write(f"channel.reset('{val}')")
+
+    def _open_channel(self, val: str) -> None:
```

Comment:
A general comment on the open/close/reset methods - strings and lists of strings (which are actually still just strings like "1101,1102") seem like they might be kind of cumbersome to use.  Could one instead use ints and allow for the user to also pass tuples of ints?  This feels like an easier way to programmatically specify channels to open/close.

Reply:
The problem here, I cannot (I mean should not) change the input types of a specific function(s). Either I should do it for the whole driver or not for the sake of clarity and coherency. Additionally, I do not see any advantage of passing 1111 instead of "1111", which differs by 2 chars which shall be auto-completed in any modern text editor, thus only one additional key stroke. Also, if user would like to do slot wise operations he/she has to pass strings anyway. 

Aside from above,  I am not in favor of this as : we have to cast every integer to string, a list of integers to strings etc as instrument only accepts strings. I do not know how much overhead this may bring.

However, if you certainly feel like this should be the way, I will do implement the changes driver-wise.

Reply:
@jdwatson do you think that `connect_column_to_rows`/`connect_row_to_columns` fulfill the request? I think they do (expecially if `disconnect` methods will be added).